source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,172 @@
+                                        update_JxW_values);
+         
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      double local_second_invariant_of_strain_rate_integral = 0.0;
+      double local_surface_area_integral = 0.0;
+
+      std::vector<std::pair<double,double>> local_second_invariant_of_strain_rate_and_corresponding_area; 
+      std::vector<SymmetricTensor<2,dim>> strain_rate (quadrature_formula.size());
+ 
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<cell->n_faces(); ++f)
+                {
+                  if (cell->at_boundary(f) && (cell->face(f)->boundary_id() == top_boundary_id))
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            if (at_upper_surface)
+              {
+                fe_face_values.reinit(cell, face_idx);
+                fe_face_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(), strain_rate);
+              
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    local_second_invariant_of_strain_rate_and_corresponding_area[q].first = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
+                    local_second_invariant_of_strain_rate_and_corresponding_area[q].second = fe_face_values.JxW(q);
+           
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate_and_corresponding_area[q].first * local_second_invariant_of_strain_rate_and_corresponding_area[q].second;
+                    // surface area over the whole surface
+                    local_surface_area_integral += local_second_invariant_of_strain_rate_and_corresponding_area[q].second;
+                  }
+              }
+          }
+      const double total_second_invariant_of_strain_rate = Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator());
+      const double total_surface_area = Utilities::MPI::sum (local_surface_area_integral, this->get_mpi_communicator());
+
+      //sort local second invariants of strain rate in size order
+      std::sort(local_second_invariant_of_strain_rate_and_corresponding_area.begin(), local_second_invariant_of_strain_rate_and_corresponding_area.end());
```

Comment:
This can still not be right. The code above basically boils down to the following (some snippets omitted):
```
  for (cell = ...)
  {
    if (something interesting is the case with 'cell')
      fill array 'local_second_invariant_of_strain_rate_and_corresponding_area' based on the quadrature points on 'cell'
  }
  ...
  std::sort (local_second_invariant_of_strain_rate_and_corresponding_area.begin(),
                 local_second_invariant_of_strain_rate_and_corresponding_area.end());
```
When you get to the call to `std::sort`, what you are sorting are the values computed on the quadrature points *of the last cell you visited*. You are not sorting the values computed on all of the cells, which I believe is what you really wanted.